addons/account/models/account_bank_statement.py

```diff
@@ -476,22 +479,69 @@ def _prepare_reconciliation_move_line(self, move, amount):
             'currency_id': statement_currency != company_currency and statement_currency.id or (st_line_currency != company_currency and st_line_currency.id or False),
             'amount_currency': amount_currency,
         }
+        if isinstance(move, self.env['account.move'].__class__):
+            aml_dict['move_id'] = move.id
+        return aml_dict
 
+    @api.multi
     def fast_counterpart_creation(self):
         """This function is called when confirming a bank statement and will allow to automatically process lines without
         going in the bank reconciliation widget. By setting an account_id on bank statement lines, it will create a journal
         entry using that account to counterpart the bank account
         """
+        payment_list = []
+        move_list = []
+        account_type_receivable = self.env.ref('account.data_account_type_receivable')
+        statement_ids = [a['statement_line_id'][0] for a in self.env['account.move.line'].read_group([('statement_line_id', 'in', self.ids)], ['statement_line_id'], ['statement_line_id'])]
         for st_line in self:
             # Technical functionality to automatically reconcile by creating a new move line
-            if st_line.account_id and not st_line.journal_entry_ids.ids:
-                vals = {
+            if st_line.account_id and not st_line.id in statement_ids:
```

Comment:
I don't know accounting well enough to understand why this improve perfs and if it's functionaly and technically equivalent to what was here before... @qdp-odoo , you're our only hope!

Reply:
It's actually a generic way to improve perfs; Basically grouping create values and using @rco-odoo 's batch create to avoid recomputing fields after *every* create and instead only doing it once.

We have a task to batch as many calls to create as possible, but as you can imagine it's more of a case-by-case thing